### DISCUSSION TITLE
Streaming data out of CARLA

### DISCUSSION THREAD

**Discussion by EricPolizzy** [carla-simulator/carla#4123] [Q&A] [answered]
Hi there,

I'd like to stream out the sensor data (e.g. collision sensor) along with vehicle control information (e.g. "Speed")  through API.
My following code is not able to capture any data. Although I used **bind(('', 8000))** specifies that the socket is reachable by any address the machine happens to have, I'm not receiving any data. 
So, I'm wondering how can I stream these data out of CARLA? Has anybody done this before?

![Screenshot 2021-04-21 104320](https://user-images.githubusercontent.com/82896556/115573733-04d4a580-a28f-11eb-8c3e-7f2eaa8a05d8.png)

Any help is greatly appreciated. Thanks in advance!

**Comment by thatdevin** [comment, marked as answer]
In your code I didn't see any clue that you were trying to fetch sensor data. The port 8000 is used for TrafficManager in Carla. Principally if you want to get sensor data, you could settle your global "data-container(like list)" in the callback function of your target sensor. In each world tick you send out this "data-container". But remember to clear it for each tick.
